src/app/games/diam/tests/diam.component.spec.ts

```diff
@@ -0,0 +1,195 @@
+        expect(testUtils.getComponent()).withContext('Component should be created').toBeTruthy();
+    });
+    it('should allow simple drops by clicking the piece and then the target', fakeAsync(async() => {
+        // given the initial state
+
+        // when clicking on a piece and then on a space
+        // then the move is made
+        await testUtils.expectClickSuccess('#piece_0_1');
+        const move: DiamMove = new DiamMoveDrop(2, DiamPiece.ZERO_SECOND);
+        await testUtils.expectMoveSuccess('#click_2', move);
+    }));
+    it('should forbid selecting a piece of the opponent', fakeAsync(async() => {
+        // given the initial state
+        // when clicking on a piece of the opponent
+        // then the corresponding error is shown
+        await testUtils.expectClickFailure('#piece_1_0', RulesFailure.MUST_CHOOSE_PLAYER_PIECE());
+    }));
+    it('should consider a piece in game click as a space click if a piece is selected for better usability', fakeAsync(async() => {
+        // given a state where there are already pieces in game
+        const state: DiamState = new DiamState([
+            [__, __, __, __, __, __, __, __],
+            [A1, __, __, __, __, __, __, __],
+            [B1, __, __, __, __, __, __, __],
+            [A1, __, __, __, __, __, __, __],
+        ], [2, 4, 3, 4], 0);
+        testUtils.setupState(state);
+        // when clicking on a remaining piece and then on a piece in game
+        // then the move is made to the corresponding space
+        await testUtils.expectClickSuccess('#piece_0_0');
+        const move: DiamMove = new DiamMoveDrop(0, DiamPiece.ZERO_FIRST);
+        await testUtils.expectMoveSuccess('#click_0_2', move);
+    }));
+    it('should forbid dropping on a full stack', fakeAsync(async() => {
+        // given a state where one stack is already full
+        const state: DiamState = new DiamState([
+            [B1, __, __, __, __, __, __, __],
+            [A1, __, __, __, __, __, __, __],
+            [B1, __, __, __, __, __, __, __],
+            [A1, __, __, __, __, __, __, __],
+        ], [2, 4, 2, 4], 0);
+        testUtils.setupState(state);
+        // when dropping a piece on the full stack
+        // then the move is rejected
+        await testUtils.expectClickSuccess('#piece_0_0');
+        await testUtils.expectMoveFailure('#click_0', DiamFailure.STACK_IS_FULL(),
+                                          new DiamMoveDrop(0, DiamPiece.ZERO_FIRST));
+    }));
+    it('should not let the user select a piece that is not available', fakeAsync(async() => {
+        // given a state where one piece is not available
+        const state: DiamState = new DiamState([
+            [A1, B1, __, __, __, __, __, __],
+            [A1, B1, __, __, __, __, __, __],
+            [A1, B1, __, __, __, __, __, __],
+            [A1, B1, __, __, __, __, __, __],
+        ], [0, 4, 0, 4], 0);
+        testUtils.setupState(state);
```

Comment:
J'utilise l'astuce du "when rendering it" pour avoir quand même un when :D !

Reply:
fait